commands/build.go

```diff
@@ -733,10 +736,16 @@ func writeMetadataFile(filename string, dt interface{}) error {
 func decodeExporterResponse(exporterResponse map[string]string) map[string]interface{} {
 	out := make(map[string]interface{})
 	for k, v := range exporterResponse {
-		dt, err := base64.StdEncoding.DecodeString(v)
-		if err != nil {
-			out[k] = v
-			continue
+		var dt []byte
+		var err error
+		if k == "result.json" {
+			dt = []byte(v)
+		} else {
```

Comment:
This area seems new. Can you expand on what this does? I'd also suggest maybe moving this section of code to a small helper function so it's more self-documenting and then you can have the function directly return `([]byte, error)` instead of using `var` and the branching logic here.

Reply:
> Can you expand on what this does?

When doing subrequests, the response looks like this: https://github.com/moby/buildkit/blob/f6eb72f2f5db07ddab89ac5e2bd3939a6444f4be/frontend/dockerui/requests.go#L100-L102

And with this PR we are setting this in the metadata file but as this is a string type https://github.com/moby/buildkit/blob/f6eb72f2f5db07ddab89ac5e2bd3939a6444f4be/frontend/dockerui/requests.go#L88, we need byte for this specific key.

I can put a comment if you wish

Reply:
Yea moving this to a comment and a small helper function I think would be good. Maybe something like `decodeExporterResponse` that contains this if/else logic or whatever naming makes more sense.